neurons/validator.py

```diff
@@ -635,6 +672,44 @@ def get_positions(self, synapse: template.protocol.GetPositions,
         synapse.error_message = error_message
         return synapse
 
+    def receive_checkpoint(self, synapse: template.protocol.ValidatorCheckpoint) -> template.protocol.ValidatorCheckpoint:
+        sender_hotkey = synapse.dendrite.hotkey
```

Comment:
map hoykey to received_checkpoint in a dictionary and reject checkpoints that were already sent. This will prevent duplicates as well as malicious validators.